agent/consul/leader_metrics.go

```diff
@@ -153,3 +177,19 @@ func initLeaderMetrics() {
 		metrics.SetGaugeWithLabels(g.Name, float32(math.NaN()), g.ConstLabels)
 	}
 }
+
+// expiresSoon checks to see if we are close enough to the cert expiring that
+// we should send out a WARN log message.
+// It returns true if the cert will expire within 28 days or 40% of the
+// certificate's total duration (whichever is shorter).
+func expiresSoon(lifetime, untilAfter time.Duration) bool {
+	defaultPeriod := 28 * (24 * time.Hour) // 28 days
+	fortyPercent := (lifetime / 10) * 4    // 40% of total duration
+
+	warningPeriod := defaultPeriod
+	if fortyPercent < defaultPeriod {
+		warningPeriod = fortyPercent
+	}
+
+	return untilAfter < warningPeriod
```

Comment:
I think we could avoid the conditional assignment by returning:

```suggestion
	return untilAfter < defaultPeriod || untilAfter < fortyPercent
```

Reply:
Interesting. I haven't seen this pattern used much in Go code, where it seems more at home in something like Python. The usual concern with these things is that it adds to the cognitive load as you have more to parse on the 1 line. But here I think it lines up pretty well with the OR rule in the comment and the function is short enough that the small bump in complexity isn't an issue.

TLDR; :+1: I like it.